// re-bulma/lib/card.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.CardFooterItem = exports.CardFooter = exports.CardHeaderIcon = exports.CardHeaderTitle = exports.CardHeader = exports.CardContent = exports.CardImage = exports.Card = undefined;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _styles = require('../build/styles');

var _styles2 = _interopRequireDefault(_styles);

var _helper = require('./helper/helper');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; } /* eslint-disable react/no-multi-comp */

var Card = exports.Card = function (_Component) {
  _inherits(Card, _Component);

  function Card() {
    _classCallCheck(this, Card);

    return _possibleConstructorReturn(this, Object.getPrototypeOf(Card).apply(this, arguments));
  }

  _createClass(Card, [{
    key: 'createClassName',
    value: function createClassName() {

      return [_styles2.default.card, this.props.isFullwidth ? _styles2.default.isFullwidth : '', this.props.className].join(' ').trim();
    }
  }, {
    key: 'render',
    value: function render() {
      return _react2.default.createElement(
        'div',
        _extends({}, (0, _helper.getCallbacks)(this.props), {
          style: this.props.style,
          className: this.createClassName()
        }),
        this.props.children
      );
    }
  }]);

  return Card;
}(_react.Component);

Card.propTypes = {
  style: _react.PropTypes.object,
  children: _react.PropTypes.any,
  className: _react.PropTypes.string,
  isFullwidth: _react.PropTypes.bool
};
Card.defaultProps = {
  style: {},
  className: ''
};

var CardImage = exports.CardImage = function (_Component2) {
  _inherits(CardImage, _Component2);

  function CardImage() {
    _classCallCheck(this, CardImage);

    return _possibleConstructorReturn(this, Object.getPrototypeOf(CardImage).apply(this, arguments));
  }

  _createClass(CardImage, [{
    key: 'createClassName',
    value: function createClassName() {

      return [_styles2.default.cardImage, this.props.className].join(' ').trim();
    }
  }, {
    key: 'render',
    value: function render() {
      return _react2.default.createElement(
        'div',
        _extends({}, (0, _helper.getCallbacks)(this.props), {
          style: this.props.style,
          className: this.createClassName()
        }),
        this.props.children
      );
    }
  }]);

  return CardImage;
}(_react.Component);

CardImage.propTypes = {
  style: _react.PropTypes.object,
  children: _react.PropTypes.any,
  className: _react.PropTypes.string
};
CardImage.defaultProps = {
  style: {},
  className: ''
};

var CardContent = exports.CardContent = function (_Component3) {
  _inherits(CardContent, _Component3);

  function CardContent() {
    _classCallCheck(this, CardContent);

    return _possibleConstructorReturn(this, Object.getPrototypeOf(CardContent).apply(this, arguments));
  }

  _createClass(CardContent, [{
    key: 'createClassName',
    value: function createClassName() {

      return [_styles2.default.cardContent, this.props.className].join(' ').trim();
    }
  }, {
    key: 'render',
    value: function render() {
      return _react2.default.createElement(
        'div',
        _extends({}, (0, _helper.getCallbacks)(this.props), {
          style: this.props.style,
          className: this.createClassName()
        }),
        this.props.children
      );
    }
  }]);

  return CardContent;
}(_react.Component);

CardContent.propTypes = {
  style: _react.PropTypes.object,
  children: _react.PropTypes.any,
  className: _react.PropTypes.string
};
CardContent.defaultProps = {
  style: {},
  className: ''
};

var CardHeader = exports.CardHeader = function (_Component4) {
  _inherits(CardHeader, _Component4);

  function CardHeader() {
    _classCallCheck(this, CardHeader);

    return _possibleConstructorReturn(this, Object.getPrototypeOf(CardHeader).apply(this, arguments));
  }

  _createClass(CardHeader, [{
    key: 'createClassName',
    value: function createClassName() {

      return [_styles2.default.cardHeader, this.props.className].join(' ').trim();
    }
  }, {
    key: 'render',
    value: function render() {
      return _react2.default.createElement(
        'header',
        _extends({}, (0, _helper.getCallbacks)(this.props), {
          style: this.props.style,
          className: this.createClassName()
        }),
        this.props.children
      );
    }
  }]);

  return CardHeader;
}(_react.Component);

CardHeader.propTypes = {
  style: _react.PropTypes.object,
  children: _react.PropTypes.any,
  className: _react.PropTypes.string
};
CardHeader.defaultProps = {
  style: {},
  className: ''
};

var CardHeaderTitle = exports.CardHeaderTitle = function (_Component5) {
  _inherits(CardHeaderTitle, _Component5);

  function CardHeaderTitle() {
    _classCallCheck(this, CardHeaderTitle);

    return _possibleConstructorReturn(this, Object.getPrototypeOf(CardHeaderTitle).apply(this, arguments));
  }

  _createClass(CardHeaderTitle, [{
    key: 'createClassName',
    value: function createClassName() {

      return [_styles2.default.cardHeaderTitle, this.props.className].join(' ').trim();
    }
  }, {
    key: 'render',
    value: function render() {
      return _react2.default.createElement(
        'p',
        _extends({}, (0, _helper.getCallbacks)(this.props), {
          style: this.props.style,
          className: this.createClassName()
        }),
        this.props.children
      );
    }
  }]);

  return CardHeaderTitle;
}(_react.Component);

CardHeaderTitle.propTypes = {
  style: _react.PropTypes.object,
  children: _react.PropTypes.any,
  className: _react.PropTypes.string
};
CardHeaderTitle.defaultProps = {
  style: {},
  className: ''
};

var CardHeaderIcon = exports.CardHeaderIcon = function (_Component6) {
  _inherits(CardHeaderIcon, _Component6);

  function CardHeaderIcon() {
    _classCallCheck(this, CardHeaderIcon);

    return _possibleConstructorReturn(this, Object.getPrototypeOf(CardHeaderIcon).apply(this, arguments));
  }

  _createClass(CardHeaderIcon, [{
    key: 'createClassName',
    value: function createClassName() {

      return [_styles2.default.cardHeaderIcon, this.props.className].join(' ').trim();
    }
  }, {
    key: 'render',
    value: function render() {

      return _react2.default.createElement(
        'a',
        _extends({}, (0, _helper.getCallbacks)(this.props), {
          style: this.props.style,
          className: this.createClassName()
        }),
        _react2.default.createElement('i', {
          className: [_styles2.default.fa, this.props.icon].join(' ').trim()
        })
      );
    }
  }]);

  return CardHeaderIcon;
}(_react.Component);

CardHeaderIcon.propTypes = {
  style: _react.PropTypes.object,
  icon: _react.PropTypes.string,
  className: _react.PropTypes.string
};
CardHeaderIcon.defaultProps = {
  style: {},
  className: ''
};

var CardFooter = exports.CardFooter = function (_Component7) {
  _inherits(CardFooter, _Component7);

  function CardFooter() {
    _classCallCheck(this, CardFooter);

    return _possibleConstructorReturn(this, Object.getPrototypeOf(CardFooter).apply(this, arguments));
  }

  _createClass(CardFooter, [{
    key: 'createClassName',
    value: function createClassName() {

      return [_styles2.default.cardFooter, this.props.className].join(' ').trim();
    }
  }, {
    key: 'render',
    value: function render() {
      return _react2.default.createElement(
        'footer',
        _extends({}, (0, _helper.getCallbacks)(this.props), {
          style: this.props.style,
          className: this.createClassName()
        }),
        this.props.children
      );
    }
  }]);

  return CardFooter;
}(_react.Component);

CardFooter.propTypes = {
  style: _react.PropTypes.object,
  children: _react.PropTypes.any,
  className: _react.PropTypes.string
};
CardFooter.defaultProps = {
  style: {},
  className: ''
};

var CardFooterItem = exports.CardFooterItem = function (_Component8) {
  _inherits(CardFooterItem, _Component8);

  function CardFooterItem() {
    _classCallCheck(this, CardFooterItem);

    return _possibleConstructorReturn(this, Object.getPrototypeOf(CardFooterItem).apply(this, arguments));
  }

  _createClass(CardFooterItem, [{
    key: 'createClassName',
    value: function createClassName() {
      return [_styles2.default.cardFooterItem, this.props.className].join(' ').trim();
    }
  }, {
    key: 'render',
    value: function render() {
      return _react2.default.createElement(
        'span',
        _extends({}, (0, _helper.getCallbacks)(this.props), {
          style: this.props.style,
          className: this.createClassName()
        }),
        this.props.children
      );
    }
  }]);

  return CardFooterItem;
}(_react.Component);

CardFooterItem.propTypes = {
  style: _react.PropTypes.object,
  children: _react.PropTypes.any,
  className: _react.PropTypes.string
};
CardFooterItem.defaultProps = {
  style: {},
  className: ''
};
