// re-bulma/lib/menu.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.MenuLink = exports.MenuList = exports.MenuLabel = exports.Menu = undefined;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _styles = require('../build/styles');

var _styles2 = _interopRequireDefault(_styles);

var _helper = require('./helper/helper');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; } /* eslint-disable react/no-multi-comp */

var Menu = exports.Menu = function (_Component) {
  _inherits(Menu, _Component);

  function Menu() {
    _classCallCheck(this, Menu);

    return _possibleConstructorReturn(this, Object.getPrototypeOf(Menu).apply(this, arguments));
  }

  _createClass(Menu, [{
    key: 'createClassName',
    value: function createClassName() {

      return [_styles2.default.menu, this.props.className].join(' ').trim();
    }
  }, {
    key: 'render',
    value: function render() {
      return _react2.default.createElement(
        'aside',
        _extends({}, (0, _helper.getCallbacks)(this.props), {
          style: this.props.style,
          className: this.createClassName()
        }),
        this.props.children
      );
    }
  }]);

  return Menu;
}(_react.Component);

Menu.propTypes = {
  style: _react.PropTypes.object,
  children: _react.PropTypes.any,
  className: _react.PropTypes.string
};
Menu.defaultProps = {
  style: {},
  className: ''
};

var MenuLabel = exports.MenuLabel = function (_Component2) {
  _inherits(MenuLabel, _Component2);

  function MenuLabel() {
    _classCallCheck(this, MenuLabel);

    return _possibleConstructorReturn(this, Object.getPrototypeOf(MenuLabel).apply(this, arguments));
  }

  _createClass(MenuLabel, [{
    key: 'createClassName',
    value: function createClassName() {

      return [_styles2.default.menuLabel, this.props.className].join(' ').trim();
    }
  }, {
    key: 'render',
    value: function render() {
      return _react2.default.createElement(
        'p',
        _extends({}, (0, _helper.getCallbacks)(this.props), {
          style: this.props.style,
          className: this.createClassName()
        }),
        this.props.children
      );
    }
  }]);

  return MenuLabel;
}(_react.Component);

MenuLabel.propTypes = {
  style: _react.PropTypes.object,
  children: _react.PropTypes.any,
  className: _react.PropTypes.string
};
MenuLabel.defaultProps = {
  style: {},
  className: ''
};

var MenuList = exports.MenuList = function (_Component3) {
  _inherits(MenuList, _Component3);

  function MenuList() {
    _classCallCheck(this, MenuList);

    return _possibleConstructorReturn(this, Object.getPrototypeOf(MenuList).apply(this, arguments));
  }

  _createClass(MenuList, [{
    key: 'createClassName',
    value: function createClassName() {

      return [_styles2.default.menuList, this.props.className].join(' ').trim();
    }
  }, {
    key: 'render',
    value: function render() {
      return _react2.default.createElement(
        'ul',
        _extends({}, (0, _helper.getCallbacks)(this.props), {
          style: this.props.style,
          className: this.createClassName()
        }),
        this.props.children
      );
    }
  }]);

  return MenuList;
}(_react.Component);

MenuList.propTypes = {
  style: _react.PropTypes.object,
  children: _react.PropTypes.any,
  className: _react.PropTypes.string
};
MenuList.defaultProps = {
  style: {},
  className: ''
};

var MenuLink = exports.MenuLink = function (_Component4) {
  _inherits(MenuLink, _Component4);

  function MenuLink() {
    _classCallCheck(this, MenuLink);

    return _possibleConstructorReturn(this, Object.getPrototypeOf(MenuLink).apply(this, arguments));
  }

  _createClass(MenuLink, [{
    key: 'createClassName',
    value: function createClassName() {
      return [this.props.isActive ? _styles2.default.isActive : '', this.props.className].join(' ').trim();
    }
  }, {
    key: 'render',
    value: function render() {
      return _react2.default.createElement(
        'a',
        _extends({}, (0, _helper.getCallbacks)(this.props), {
          href: this.props.href,
          style: this.props.style,
          className: this.createClassName()
        }),
        this.props.children
      );
    }
  }]);

  return MenuLink;
}(_react.Component);

MenuLink.propTypes = {
  style: _react.PropTypes.object,
  children: _react.PropTypes.any,
  className: _react.PropTypes.string,
  href: _react.PropTypes.string,
  isActive: _react.PropTypes.bool
};
MenuLink.defaultProps = {
  style: {},
  className: ''
};
