// re-bulma/lib/panel.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.PanelBlock = exports.PanelTabs = exports.PanelHeading = exports.Panel = undefined;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _styles = require('../build/styles');

var _styles2 = _interopRequireDefault(_styles);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; } /* eslint-disable react/no-multi-comp */

var Panel = exports.Panel = function (_Component) {
  _inherits(Panel, _Component);

  function Panel() {
    _classCallCheck(this, Panel);

    return _possibleConstructorReturn(this, Object.getPrototypeOf(Panel).apply(this, arguments));
  }

  _createClass(Panel, [{
    key: 'createClassName',
    value: function createClassName() {
      return [_styles2.default.panel, this.props.className].join(' ').trim();
    }
  }, {
    key: 'render',
    value: function render() {
      return _react2.default.createElement(
        'nav',
        _extends({}, this.props, { className: this.createClassName() }),
        this.props.children
      );
    }
  }]);

  return Panel;
}(_react.Component);

Panel.propTypes = {
  children: _react.PropTypes.any,
  className: _react.PropTypes.string
};
Panel.defaultProps = {
  className: ''
};

var PanelHeading = exports.PanelHeading = function (_Component2) {
  _inherits(PanelHeading, _Component2);

  function PanelHeading() {
    _classCallCheck(this, PanelHeading);

    return _possibleConstructorReturn(this, Object.getPrototypeOf(PanelHeading).apply(this, arguments));
  }

  _createClass(PanelHeading, [{
    key: 'createClassName',
    value: function createClassName() {
      return [_styles2.default.panelHeading, this.props.className].join(' ').trim();
    }
  }, {
    key: 'render',
    value: function render() {
      return _react2.default.createElement(
        'p',
        _extends({}, this.props, { className: this.createClassName() }),
        this.props.children
      );
    }
  }]);

  return PanelHeading;
}(_react.Component);

PanelHeading.propTypes = {
  children: _react.PropTypes.any,
  className: _react.PropTypes.string
};
PanelHeading.defaultProps = {
  className: ''
};

var PanelTabs = exports.PanelTabs = function (_Component3) {
  _inherits(PanelTabs, _Component3);

  function PanelTabs() {
    _classCallCheck(this, PanelTabs);

    return _possibleConstructorReturn(this, Object.getPrototypeOf(PanelTabs).apply(this, arguments));
  }

  _createClass(PanelTabs, [{
    key: 'createClassName',
    value: function createClassName() {
      return [_styles2.default.panelTabs, this.props.className].join(' ').trim();
    }
  }, {
    key: 'render',
    value: function render() {
      return _react2.default.createElement(
        'p',
        _extends({}, this.props, { className: this.createClassName() }),
        this.props.children
      );
    }
  }]);

  return PanelTabs;
}(_react.Component);

PanelTabs.propTypes = {
  children: _react.PropTypes.any,
  className: _react.PropTypes.string
};
PanelTabs.defaultProps = {
  className: ''
};

var PanelBlock = exports.PanelBlock = function (_Component4) {
  _inherits(PanelBlock, _Component4);

  function PanelBlock() {
    _classCallCheck(this, PanelBlock);

    return _possibleConstructorReturn(this, Object.getPrototypeOf(PanelBlock).apply(this, arguments));
  }

  _createClass(PanelBlock, [{
    key: 'createClassName',
    value: function createClassName() {
      return [_styles2.default.panelBlock, this.props.isActive ? _styles2.default.isActive : '', this.props.className].join(' ').trim();
    }
  }, {
    key: 'render',
    value: function render() {
      return _react2.default.createElement(
        'span',
        _extends({}, this.props, { className: this.createClassName() }),
        this.props.icon ? _react2.default.createElement(
          'span',
          { className: _styles2.default.panelIcon },
          _react2.default.createElement('i', { className: [_styles2.default.fa, this.props.icon].join(' ') })
        ) : null,
        this.props.children
      );
    }
  }]);

  return PanelBlock;
}(_react.Component);

PanelBlock.propTypes = {
  children: _react.PropTypes.any,
  className: _react.PropTypes.string,
  isActive: _react.PropTypes.bool,
  icon: _react.PropTypes.string
};
PanelBlock.defaultProps = {
  className: ''
};
