// re-bulma/lib/styles.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _templateObject = _taggedTemplateLiteral(['', ''], ['', '']);

var _csjs = require('csjs');

var _csjs2 = _interopRequireDefault(_csjs);

var _insertCss = require('insert-css');

var _insertCss2 = _interopRequireDefault(_insertCss);

var _box = require('./styles/box');

var _box2 = _interopRequireDefault(_box);

var _button = require('./styles/button');

var _button2 = _interopRequireDefault(_button);

var _columns = require('./styles/columns');

var _columns2 = _interopRequireDefault(_columns);

var _form = require('./styles/form');

var _form2 = _interopRequireDefault(_form);

var _icon = require('./styles/icon');

var _icon2 = _interopRequireDefault(_icon);

var _label = require('./styles/label');

var _label2 = _interopRequireDefault(_label);

var _table = require('./styles/table');

var _table2 = _interopRequireDefault(_table);

var _tabs = require('./styles/tabs');

var _tabs2 = _interopRequireDefault(_tabs);

var _tag = require('./styles/tag');

var _tag2 = _interopRequireDefault(_tag);

var _title = require('./styles/title');

var _title2 = _interopRequireDefault(_title);

var _content = require('./styles/content');

var _content2 = _interopRequireDefault(_content);

var _progress = require('./styles/progress');

var _progress2 = _interopRequireDefault(_progress);

var _image = require('./styles/image');

var _image2 = _interopRequireDefault(_image);

var _message = require('./styles/message');

var _message2 = _interopRequireDefault(_message);

var _notification = require('./styles/notification');

var _notification2 = _interopRequireDefault(_notification);

var _card = require('./styles/card');

var _card2 = _interopRequireDefault(_card);

var _level = require('./styles/level');

var _level2 = _interopRequireDefault(_level);

var _heading = require('./styles/heading');

var _heading2 = _interopRequireDefault(_heading);

var _modal = require('./styles/modal');

var _modal2 = _interopRequireDefault(_modal);

var _mediaObject = require('./styles/media-Object');

var _mediaObject2 = _interopRequireDefault(_mediaObject);

var _menu = require('./styles/menu');

var _menu2 = _interopRequireDefault(_menu);

var _nav = require('./styles/nav');

var _nav2 = _interopRequireDefault(_nav);

var _pagination = require('./styles/pagination');

var _pagination2 = _interopRequireDefault(_pagination);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _taggedTemplateLiteral(strings, raw) { return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

var styles = {
  box: _box2.default,
  button: _button2.default,
  columns: _columns2.default,
  form: _form2.default,
  icon: _icon2.default,
  label: _label2.default,
  table: _table2.default,
  tag: _tag2.default,
  tabs: _tabs2.default,
  title: _title2.default,
  content: _content2.default,
  progress: _progress2.default,
  message: _message2.default,
  image: _image2.default,
  notification: _notification2.default,
  card: _card2.default,
  level: _level2.default,
  heading: _heading2.default,
  modal: _modal2.default,
  mediaObject: _mediaObject2.default,
  menu: _menu2.default,
  nav: _nav2.default,
  pagination: _pagination2.default
};

var defaultOptions = {
  prepend: true,
  components: {
    box: true,
    button: true,
    columns: true,
    form: true,
    icon: true,
    label: true,
    table: true,
    tabs: true,
    tag: true,
    title: true,
    content: true,
    progress: true,
    message: true,
    image: true,
    notification: true,
    card: true,
    level: true,
    heading: true,
    modal: true,
    mediaObject: true,
    menu: true,
    nav: true,
    pagination: true
  }
};

var instance = null;

var Styles = function () {
  function Styles() {
    _classCallCheck(this, Styles);

    if (!instance) instance = this;
    return instance;
  }

  _createClass(Styles, [{
    key: 'getStyles',
    value: function getStyles() {
      return this.styles;
    }
  }, {
    key: 'initialize',
    value: function initialize() {
      var options = arguments.length <= 0 || arguments[0] === undefined ? defaultOptions : arguments[0];

      var combinedStyles = Object.keys(options.components).filter(function (key) {
        return options.components[key];
      }).reduce(function (pre, cur) {
        return pre + styles[cur]();
      }, '');
      console.time('1');
      this.styles = (0, _csjs2.default)(_templateObject, combinedStyles);
      console.timeEnd('1');
      console.dir(this.styles);
      console.time('2');
      (0, _insertCss2.default)(_csjs2.default.getCss(this.styles), { prepend: options.prepend });
      console.timeEnd('2');
    }
  }]);

  return Styles;
}();

exports.default = Styles;
